### PR TITLE
Attempt to improve CI caching

### PR DESCRIPTION
**Pull request by max-sixty**
Currently about 40% of the time is taken by installing things, hopefully we can cut that down

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -74,9 +77,18 @@ jobs:
           python-version: ${{ matrix.python-version }}
           use-only-tar-bz2: true
 
+      - name: Cache conda env
+        id: cache-env
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.CONDA }}/envs
```
Comment (dcherian) [inline .github/workflows/ci.yaml:84]: Maybe 
```suggestion
          path: ${{ env.CONDA_PREFIX }}
```

`env.CONDA` is empty looking at the logs...

image

Comment (max-sixty) [reply .github/workflows/ci.yaml]: Thanks. Weirdly it's listed in https://github.com/conda-incubator/setup-miniconda#caching-environments...

Comment (max-sixty) [reply .github/workflows/ci.yaml]: Sorry, actually where do you see that it's empty?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -74,9 +77,18 @@ jobs:
           python-version: ${{ matrix.python-version }}
           use-only-tar-bz2: true
 
+      - name: Cache conda env
+        id: cache-env
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.CONDA_PREFIX }}
```
Comment (dcherian) [inline .github/workflows/ci.yaml:84]:
```suggestion
          path: /usr/share/miniconda/envs/xarray-tests
```

Comment (dcherian) [reply .github/workflows/ci.yaml]: Just to check

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (max-sixty) [reply .github/workflows/ci.yaml]: Ah, right, yes, `/envs`

**Comment by dcherian** [comment]
The output under "Version Info"  lists these environment variables
```
CONDA: /usr/share/miniconda
CONDA_DEFAULT_ENV: xarray-tests
CONDA_ENV_FILE: ci/requirements/environment.yml
CONDA_EXE: /usr/share/miniconda/bin/conda
CONDA_PKGS_DIR: /home/runner/conda_pkgs_dir
CONDA_PREFIX: /usr/share/miniconda/envs/xarray-tests
CONDA_PREFIX_1: /usr/share/miniconda
CONDA_PROMPT_MODIFIER: 
CONDA_PYTHON_EXE: /usr/share/miniconda/bin/python
CONDA_ROOT: /usr/share/miniconda
CONDA_SHLVL: 2
```

I don't understand why these variables aren't being set earlier. It's almost as if the environment isn't being activated

**Comment by max-sixty** [comment]
Yeah, that is weird. If the `CONDA_PREFIX` doesn't work, we could do the very hacky thing of just using `/usr/share/miniconda/envs/xarray-tests`...

**Comment by dcherian** [comment]
Nice work. This is working well!

image


We could also skip installing the "error reporter" but its only 2s.

**Comment by max-sixty** [comment]
Nice! Thanks for your help on the weird env issue!

**Pull request merged**